src/s3path.jl

```diff
@@ -0,0 +1,235 @@
+struct S3Path <: AbstractPath
+    segments::Tuple{Vararg{String}}
+    root::String
+    drive::String
+    dir::Bool
+    config::AWSConfig
+end
+
+function S3Path()
+    config = aws_config()
+    account_id = aws_account_number(config)
+    region = config[:region]
+
+    return S3Path(
+        (),
+        "/",
+        "s3://$account_id-$region",
+        true,
+        config,
+    )
+end
+
+function S3Path(str::AbstractString; config::AWSConfig=aws_config())
+    str = String(str)
+    root = ""
+    path = ()
+    dir = true
```

Comment:
Can we call this `isdirectory`?
`dir` to me sounds like it might be the directory part of the name.

Reply:
👌